### PR TITLE
Pipedream source documentation

### PR DESCRIPTION
**Pull request by ameypv-rudder**
## What do these changes do?

> Added Pipedream source documentation

## Type of documentation

- [x] New documentation
- [ ] Documentation update
- [ ] Hotfix

### CODE REVIEW

File: docs/sources/event-streams/cloud-apps/pipedream.mdx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    "song": "Take Five"
+  }
+}
+```
+
+<br />
+
+### Event transformation
+
+RudderStack ingests the Pipedream events after converting them into the RudderStack event format. It also maps the following properties from the Pipedream event payload to the RudderStack properties:
```
Comment (anantjain45823) [inline docs/sources/event-streams/cloud-apps/pipedream.mdx:71]:
```suggestion
RudderStack ingests the Pipedream events and checks for userId or anonymousId and type (identify ,track etc.) . If both are present it sees it as a rudderstack format payload other wise converts it into rudderstack payload format according to the below mapping:
```

File: docs/sources/event-streams/cloud-apps/pipedream.mdx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+| Pipedream property  | RudderStack property|
```
Comment (anantjain45823) [inline docs/sources/event-streams/cloud-apps/pipedream.mdx:73]:
```suggestion
| RudderStack property  | Value | 
```

File: docs/sources/event-streams/cloud-apps/pipedream.mdx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+### Event transformation
+
+RudderStack ingests the Pipedream events after converting them into the RudderStack event format. It also maps the following properties from the Pipedream event payload to the RudderStack properties:
+
+| Pipedream property  | RudderStack property| 
+| :------------------ | :------------- |
+| `pipedream_source_event`  | `message.event`    | 
+| `track`    | `message.type`        | 
+| input payload       | `message.properties`|
```
Comment (anantjain45823) [inline docs/sources/event-streams/cloud-apps/pipedream.mdx:77]:
```suggestion
| :------------------ | :------------- |
| `message.event`  | `pipedream_source_event`    | 
|  `message.type`   |  `track`       | 
| `message.properties`     | pipedream input payload |
```

File: docs/sources/event-streams/cloud-apps/pipedream.mdx [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+Follow these steps to set up your Pipedream source in the RudderStack dashboard:
+
+1. Go to your [RudderStack dashboard](https://app.rudderstack.com/) and click **Add Source**. From the list of **Event Streams** sources, select **Pipedream**.
+2. Assign a name to your source and click **Continue**.
+3. Your Pipedream source is now configured. Note the **Webhook URL** in the **Settings** tab, as shown:
+
+<img src="../../../assets/event-stream-sources/pipedream-webhook.png" alt="Pipedream source webhook URL" />
+
+4. Log in to your [Pipedream dashboard](https://pipedream.com/workflows/) and set up your workflow.
+5. After <a href="https://pipedream.com/new?tutorial=1">adding a trigger</a>, click the **+** icon to add a step. Then, select the **HTTP / Webhook** option and choose **Send any HTTP Request**, as shown:
```
Comment (anantjain45823) [inline docs/sources/event-streams/cloud-apps/pipedream.mdx:22]: Can we accommodate this statement as well. User can select any other HTTP type as well but rudderstack would take only POST requests made to Ruderstack.

File: docs/sources/event-streams/cloud-apps/pipedream.mdx [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+<img src="../../../assets/event-stream-sources/pipedream-http-webhook.png" alt="Pipedream add a step" />
+
+6. In the **HTTP Request Configuration** section, choose **POST** and enter the RudderStack webhook URL obtained in Step 3 above. Also, configure the other request configuration settings as required.
```
Comment (ameypv-rudder) [inline docs/sources/event-streams/cloud-apps/pipedream.mdx:26]:
```suggestion
6. In the **HTTP Request Configuration** section, choose **POST** as the request type and enter the RudderStack webhook URL obtained in Step 3 above. Also, configure the other request configuration settings as required.
```

File: docs/sources/event-streams/cloud-apps/pipedream.mdx (hunk 4 above)
Comment (ameypv-rudder) [reply docs/sources/event-streams/cloud-apps/pipedream.mdx:22]: I've added this as a note below Step 6.

Comment (anantjain45823) [reply docs/sources/event-streams/cloud-apps/pipedream.mdx:22]: Yeah I saw but it is not only for this source it is in genral. So need to update Note according to it

**Review by anantjain45823 — APPROVED** [review]

File: docs/sources/event-streams/cloud-apps/pipedream.mdx [hunk 6]
```diff
@@ -0,0 +1,81 @@
+---
+title: "Pipedream"
+description: Ingesting your event data from Pipedream into RudderStack.
+aliases: ["Pipedream source"]
+---
+
+[Pipedream](https://pipedream.com/) lets you build and automate processes that connect APIs. It supports open source triggers and actions for hundreds of integrations.
+
+This guide will help you set up Pipedream as a source in RudderStack.
```
Comment (ishachopra0401) [inline docs/sources/event-streams/cloud-apps/pipedream.mdx:9]: can skip if seems fine to you as well

File: docs/sources/event-streams/cloud-apps/pipedream.mdx [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+RudderStack only accepts the incoming HTTP requests of type <strong>POST</strong>.
+</div>
+
+<img src="../../../assets/event-stream-sources/pipedream-custom-http-request.png" alt="Pipedream custom HTTP request" />
+
+7. Click **Deploy** to finish setting up the workflow.
+
+## How RudderStack creates the event payload
+
+RudderStack supports all the <Link to="/event-spec/standard-events/">standard events</Link> in the specified format. Otherwise, it transforms the incoming data to a <Link to="/event-spec/standard-events/track/">`track`</Link> payload.
```
Comment (ishachopra0401) [inline docs/sources/event-streams/cloud-apps/pipedream.mdx:38]:
```suggestion
RudderStack supports all the <Link to="/event-spec/standard-events/">standard events</Link> in the specified format. For the rest of the events, it transforms the incoming data to a <Link to="/event-spec/standard-events/track/">`track`</Link> payload.
```

File: docs/sources/event-streams/cloud-apps/pipedream.mdx [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+    "artist": "Artist One",
+    "genre": "Jazz",
+    "song": "Take Five"
+  }
+}
+```
+
+<br />
+
+### Event transformation
```
Comment (ishachopra0401) [inline docs/sources/event-streams/cloud-apps/pipedream.mdx:73]: Should we move **## Event transformation** above **## How RudderStack creates the event payload** similar to the other source docs? It gives a little more clarity by going through the mappings first and then seeing the payload. WDYT?

Comment (ameypv-rudder) [reply docs/sources/event-streams/cloud-apps/pipedream.mdx:73]: Agreed! Made the change.

File: docs/sources/event-streams/cloud-apps/pipedream.mdx (hunk 7 above)
Comment (ameypv-rudder) [reply docs/sources/event-streams/cloud-apps/pipedream.mdx:38]: Slightly tweaked to "For the other events..." to make it a little less wordy. WDYT?

File: docs/sources/event-streams/cloud-apps/pipedream.mdx (hunk 6 above)
Comment (ameypv-rudder) [reply docs/sources/event-streams/cloud-apps/pipedream.mdx:9]: Agreed - made the change.

**Review by ishachopra0401 — APPROVED** [review]

**Pull request merged**